denguefever_tw/dengue_linebot/views.py

```diff
@@ -316,3 +316,53 @@ def suggestion_list(request):
 def gov_report_list(request):
     context = {'gov_reports': GovReport.objects.all().order_by('-report_time')}
     return render(request, 'dengue_linebot/gov_report_list.html', context)
+
+
+@login_required
+def push_msg_form(request):
+    context = {'areas': MinArea.objects.all()}
+    return render(request, 'dengue_linebot/push_msg.html', context)
+
+
+@login_required
+def push_msg_result(request):
+    areas_id = request.POST.getlist('msg_to')
+    content = request.POST['content']
+    img = request.POST['img']
+    error_msgs = list()
+    push_logs = list()
+
+    if not areas_id:
+        error_msgs.append('You do not choose any area!')
+    if not content and not img:
+        error_msgs.append('You do not write any content or image!')
+
+    if not error_msgs:
+        for area_id in areas_id:
+            users = LineUser.objects.filter(location=MinArea.objects.get(area_id=area_id))
+            if users:
+                push_logs.extend(_push_msg(users=users, text=content, img=img))
+
+    return render(request, 'dengue_linebot/push_msg_result.html', {
+        'error_msgs':error_msgs,
+        'push_logs':push_logs
+    })
+
+
+def _push_msg(users, text, img):
+    splited_users_lists = [users[i:i + 150] for i in range(0, len(users), 150)]
```

Comment:
The number 150 should be stored in a constant variable (Use name with upper letters) instead of hard coding.